conans/model/conan_file.py

```diff
@@ -142,6 +142,9 @@ class ConanFile(object):
     # Run in windows bash
     win_bash = None
 
+    # For test_package, reference being tested
+    tested_reference_str = None
```

Comment:
As this is very specific to only test_package, I am fine if this is dynamic and not a class attribute (but fine with it too)